### PR TITLE
WIP: move ICS 20 code to 20-transfer

### PR DESCRIPTION
**Pull request by chengwenxi**
Move ICS 20 code to 20-transfer

### CODE REVIEW

File: x/ibc/20-transfer/client/cli/tx.go [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+			// parse coin trying to be sent
+			coin, err := sdk.ParseCoin(args[3])
+			if err != nil {
+				return err
+			}
+
+			source := false
+			if viper.GetBool(FlagSource) {
+				source = true
+			}
```
Comment (dreamer-zq) [inline x/ibc/20-transfer/client/cli/tx.go:58]: source=viper.GetBool(FlagSource)

File: x/ibc/20-transfer/kepper/keeper.go [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+	}
+
+	if data.Source {
+		// mint tokens
+
+		// check the denom prefix
+		prefix := fmt.Sprintf("%s/%s", destPort, destChannel)
+		if !strings.HasPrefix(data.Denomination, prefix) {
+			return sdk.NewError(sdk.CodespaceType(types.DefaultCodespace), types.CodeIncorrectDenom, "incorrect denomination")
+		}
```
Comment (dreamer-zq) [inline x/ibc/20-transfer/kepper/keeper.go]: duplicate code, can be refactored into a method

File: x/ibc/20-transfer/types/msgs.go [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+const (
+	TypeMsgTransfer = "transfer"
+)
+
+type MsgTransfer struct {
+	SrcPort      string         `json:"src_port" yaml:"src_port"`
+	SrcChannel   string         `json:"src_channel" yaml:"src_channel"`
+	Denomination string         `json:"denomination" yaml:"denomination"`
+	Amount       sdk.Int        `json:"amount" yaml:"amount"`
+	Sender       sdk.AccAddress `json:"sender" yaml:"sender"`
```
Comment (dreamer-zq) [inline x/ibc/20-transfer/types/msgs.go:16]: Type of sender should be string ?

Comment (chengwenxi) [reply x/ibc/20-transfer/types/msgs.go:16]: > Type of sender should be string ?

it's string in packet.

**Pull request closed without merging**